File: src/my_project_typescript/easy_problems_typescript/from151to200/minCostClimbingDP.ts
function minCostClimbingStairs(cost: number[]): number {
    let l: number = cost.length
    let dp: Array<number> = Array(l+2).fill(0)

    for (let i = 3; i < l+2; i++){
        dp[i] = Math.min(dp[i-1]+cost[i-2], dp[i-2]+cost[i-3])
    }

    return dp[l+1]

};

File: src/my_project/easy_problems/from251to300/min_cost_climbing_dp.py
from typing import List 

class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        l = len(cost)
        dp = [0]*(l+2)
        for i in range(3,l+2):
            dp[i] = min(dp[i-1]+cost[i-2],dp[i-2]+cost[i-3])
        return dp[l+1]